### ISSUE TITLE
Crash on undo after jump

### ISSUE THREAD

**Issue by anusser** [xournalpp/xournalpp#5194]
### Operating System

Linux

### (Linux only) Distribution

Manjaro

### (Linux only) Desktop Environment

KDE

### (Linux Only) Display Server

X.Org

### Installation Method

Manjaro Repository

### Xournal++ Version

1.2.1

### libgtk Version

3.24.38

### Bug Description

Xournal++ crashes when undoing an action after jumping to another location. See "Steps to Reproduce" for the exact steps.

### Expected Behaviour

Well... not crashing? ;)

### Steps to Reproduce

1. Open some pdf for annotation that has hyperlinks (e.g. in latex to a bibliography)
2. Draw something with the pen
3. Use the "Select linear PDF text" tool, click a reference, and click "Scroll to page XX"
4. Now hit the undo button (yellow back arrow)
5. After step 4, Xournal++ crashes on me every time

### Additional Context

Relevant output when crashing:

(com.github.xournalpp.xournalpp:99627): Gtk-CRITICAL **: 18:26:36.848: gtk_tree_model_iter_children: assertion 'GTK_IS_TREE_MODEL (tree_model)' failed

(com.github.xournalpp.xournalpp:99627): GLib-GObject-CRITICAL **: 18:26:36.848: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
xournalpp: /usr/src/debug/xournalpp/xournalpp/src/core/control/jobs/RenderJob.cpp:48: void RenderJob::rerenderRectangle(const xoj::util::Rectangle<double>&): Assertion `view->buffer.isInitialized()' failed.

(com.github.xournalpp.xournalpp:99627): xopp-WARNING **: 18:27:11.438: [Crash Handler] Crashed with signal 6

**Comment by anusser** [comment]
As always: Thank you very much for creating such wonderful software and I'm happy to help debugging and testing potential fixes. :)

**Comment by rolandlo** [comment]
Duplicate of #5046. Thanks for offering help. We'll let you know when we need help testing.
